app/javascript/pages/Wishlists/Index.tsx

```diff
@@ -17,54 +14,52 @@ import { WithTooltip } from "$app/components/WithTooltip";
 
 import placeholder from "$assets/images/placeholders/wishlists.png";
 
-type Wishlist = {
+export type Wishlist = {
   id: string;
   name: string;
   url: string;
   product_count: number;
   discover_opted_out: boolean;
 };
 
-const WishlistsPage = ({
-  wishlists: preloadedWishlists,
-  reviews_page_enabled,
-  following_wishlists_enabled,
-}: {
+type Props = {
   wishlists: Wishlist[];
   reviews_page_enabled: boolean;
   following_wishlists_enabled: boolean;
-}) => {
+};
+
+export default function WishlistsPage() {
+  const {
+    wishlists: preloadedWishlists,
+    reviews_page_enabled,
+    following_wishlists_enabled,
+  } = cast<Props>(usePage().props);
+
   const [wishlists, setWishlists] = React.useState<Wishlist[]>(preloadedWishlists);
```

Comment:
You shouldn't need to manage this internally, this should update from the backend only now